edgelet/edgelet-http/src/lib.rs

```diff
@@ -217,9 +223,37 @@ where
 
         let protocol = Arc::new(protocol);
 
+        let (lock, limit) = if let ConcurrencyThrottling::Limited(limit) = concurrency {
```

Comment:
`limit` is only used when `lock` is `Some`, so put it inside the same `Option`

```rust
let limit = if let ConcurrencyThrottling::Limited(limit) = concurrency {
    Some((Arc::new(AtomicU32::new(*limit)), *limit))
} else {
    None
};

...

if let Some((lock, limit)) = lock.clone() {
```

Reply:
Good point, it looks nicer. Fixed